doc/governance/proposals.rst

```diff
@@ -46,52 +46,200 @@ For transparency and auditability, all governance operations (including votes) a
+.. code-block:: json
 
-Additional detail is available from the ``--help`` option. You can also find the script in a checkout of CCF:
+    {
+      "actions": [
+        {
+          "name": "set_node_certificate_validity",
+          "args": {
+            "node_id": "ba9faac9683f7854c2cf0a97f57e63c260bf8d06f8183772c5655093c0af6e19",
+            "valid_from": "220112101937Z",
+            "validity_period_days": 366
+          }
+        }
+      ]
+    }
 
-.. code-block:: bash
+.. code-block:: json
 
-    $ python CCF/python/ccf/proposal_generator.py
+    {
+      "actions": [
+        {
+          "name": "transition_node_to_trusted",
+          "args": {
+            "node_id": "ba9faac9683f7854c2cf0a97f57e63c260bf8d06f8183772c5655093c0af6e19",
+            "valid_from": "220101120000Z"
+          }
+        },
+        {
+          "name": "transition_node_to_trusted",
+          "args": {
+            "node_id": "5d5b09f6dcb2d53a5fffc60c4ac0d55fabdf556069d6631545f42aa6e3500f2e",
+            "valid_from": "220101120000Z"
+          }
+        },
+        {
+          "name": "transition_service_to_open"
+        }
+      ]
+    }
 
-Some of these subcommands require additional arguments, such as the node ID or user certificate to add to the service. Additional options allow the generated votes and proposals to be redirected to other files or pretty-printed:
+Most HTTP client libraries and tools should have functionality for constructing and providing these JSON objects, and constitutions should be written to provide clear validation errors if a proposal is malformed.
 
-.. code-block:: bash
+A ballot's body contains a JS function which evaluates a given proposal, embedded inside a JSON object.
+These may try to confirm the precise content equality of the proposal they are considering, or put some constraints on its parameters.
+They could also be simple positive/negative votes, in a model where members fetch and validate a proposal offline before submitting their votes.
+Some example ballots which could apply to the proposals above:
 
-    $ python -m ccf.proposal_generator transition_node_to_trusted 6d566123a899afaea977c5fc0f7a2a9fef33f2946fbc4abefbc3e10ee597343f 211019154318Z
-    SUCCESS | Writing proposal to ./trust_node_proposal.json
-    SUCCESS | Wrote vote to ./trust_node_vote_for.json
+.. code-block:: json
 
-    $ cat trust_node_proposal.json
-    {"actions": [{"name": "transition_node_to_trusted", "args": {"node_id": "6d566123a899afaea977c5fc0f7a2a9fef33f2946fbc4abefbc3e10ee597343f", "valid_from": "211019154318Z"}}]}
+    {
+      "ballot": "export function vote (rawProposal, proposerId)\n
```

Comment:
Another docs styling choice - I've added real new-lines next to the embedded escaped new-lines. This matches how these string are pretty-printed in JS, and I think is far more readable, at the cost of not being copy-pastable as a valid JSON object.